packages/client/src/mud/setup.ts

```diff
@@ -45,11 +47,27 @@ export async function setup() {
     setInterval(requestDrip, 20000);
   }
 
+  if (!signer) throw new Error("No signer");
+  if (!signer.provider) throw new Error("No provider connected to the signer");
+
+  // Create a World contract instance
+  const worldContract = IWorld__factory.connect(
+    networkConfig.worldAddress,
+    signer
+  );
```

Comment:
I did a similar thing in emojimon branch, but opted not to do the signer check, thinking that folks might wanna build a spectator-mode thing?

https://github.com/latticexyz/v2sandbox/pull/17/files#diff-30525edeafc7d169366fa321b43901a7e94518ff244c36942f38618c4446f6bcR56-R59

(that said, I guess we ~always instantiate with a burner wallet and start a drip, so we theoretically should ~always have a signer...)

Reply:
that works! I guess we could add an error in fastTxExecute if no signer is connected?